.circleci/config.yml

```diff
@@ -145,34 +151,107 @@ jobs:
       - store_artifacts:
           path: test-results
 
-  unpackaged-sanity:
+  linux-tarball-smoke-tests:
+    <<: *defaults
+    parameters:
+      # If true, the tests will use the LTS version of node. If false, the latest version of node will be used instead.
+      lts:
+        type: boolean
+        default: true
+    steps:
+      - attach_workspace:
+          at: ~/repo
+
+      # Upgrade to whatever version of Node we need.
+      - node/install:
+          install-npm: false
+          lts: << parameters.lts >>
+
+      # Log the version of node we're using post-upgrade.
+      - run: node -v
+
+      - run:
+          name: "Install sfdx-cli"
+          command: npm install -g sfdx-cli
+
+      # Build and install our tarball.
+      - run:
+          name: "Build and install tarball"
+          command: |
+            yarn pack --filename tarball.tgz
+            # Pipe in a 'y' to simulate agreeing to install an unsigned plugin. Use a URI consisting of the file's full path.
+            echo y | sfdx plugins:install file://`pwd`/tarball.tgz
+
+      # Persist the workspace, so the tarball is visible to subsequent jobs in this flow.
+      - persist_to_workspace:
+          root: ~/repo
+          paths: .
+
+      # Create a directory for the test output.
+      - run: mkdir test-results
+
+      # Run the smoke tests. If none of them explode, we can reasonably assume that the plugin is safe.
+      - run:
+          name: "Smoke Tests"
+          command: smoke-tests/smoke-test.sh sfdx
```

Comment:
This file shows up as `Empty file.` for me on this review. Any idea why?

Reply:
I didn't change the contents of the file; I just changed its default permissions so the CI can execute it, which I was seeing it struggle to do.